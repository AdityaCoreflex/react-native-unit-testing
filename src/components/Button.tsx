// components/Button.tsx
import React from 'react';
import { Alert, StyleSheet, Text, TouchableOpacity } from 'react-native';

type ButtonProps = {
  onPress?: () => void;
  label?: string;
};

const Button = ({ onPress, label = 'Press Me' }: ButtonProps) => {
  return (
    <TouchableOpacity
      testID="sample-button"
      style={styles.button}
      onPress={onPress || (() => Alert.alert('Button Pressed'))}
    >
      <Text style={styles.buttonText}>{label}</Text>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  button: {
    backgroundColor: '#007AFF',
    padding: 10,
    marginVertical: 10,
    borderRadius: 8,
    alignItems: 'center',
  },
  buttonText: {
    color: '#fff',
  },
});
