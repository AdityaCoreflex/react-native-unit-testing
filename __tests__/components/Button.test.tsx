import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from '../../src/components/Button';

describe('Button Component', () => {
  it('renders with default label', () => {
    const { getByTestId } = render(<Button />);
    const button = getByTestId('sample-button');
    expect(button).toBeTruthy();
  });

  it('renders with custom label', () => {
    const { getByText } = render(<Button label="Submit" />);
    expect(getByText('Submit')).toBeTruthy();
  });

  it('calls onPress when pressed', () => {
    const mockFn = jest.fn();
    const { getByTestId } = render(<Button onPress={mockFn} />);
    fireEvent.press(getByTestId('sample-button'));
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
  it('default snapshot testing', () => {
    const tree = render(<Button />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('matches snapshot after clicking', () => {
    const mockFn = jest.fn();
    const { getByTestId, toJSON } = render(<Button onPress={mockFn} />);
    fireEvent.press(getByTestId('sample-button'));
    expect(toJSON()).toMatchSnapshot();
  });
});
