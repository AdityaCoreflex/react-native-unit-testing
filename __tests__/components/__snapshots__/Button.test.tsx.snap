// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component default snapshot testing 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "marginVertical": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="sample-button"
>
  <Text
    style={
      {
        "color": "#fff",
      }
    }
  >
    Press Me
  </Text>
</View>
`;

exports[`Button Component matches snapshot after clicking 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "marginVertical": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="sample-button"
>
  <Text
    style={
      {
        "color": "#fff",
      }
    }
  >
    Press Me
  </Text>
</View>
`;
